github/azure-self-hosted-runners/tf/main.tf

```diff
@@ -0,0 +1,170 @@
+resource "azurerm_resource_group" "garm_rg" {
+  name     = var.resource_group_name
+  location = var.location
+}
+
+resource "azurerm_user_assigned_identity" "garm_id" {
+  name                = "garm"
+  location            = azurerm_resource_group.garm_rg.location
+  resource_group_name = azurerm_resource_group.garm_rg.name
+}
+
+data "azurerm_subscription" "current" {
+}
+
+data "azurerm_client_config" "current" {
+}
+
+resource "random_string" "suffix" {
+  length  = 6
+  lower   = true
+  upper   = false
+  special = false
+}
+
+resource "azurerm_role_assignment" "garm_role_as" {
+  scope                = data.azurerm_subscription.current.id
+  role_definition_name = "Contributor"
+  principal_id         = azurerm_user_assigned_identity.garm_id.principal_id
+}
+
+resource "azurerm_storage_account" "garm_sa" {
+  name                     = "garm${random_string.suffix.result}"
```

Comment:
Wouldn't this always trigger a redeployment?

Reply:
no, you can define criterias to trigger re-creation, but the random resource will be saved in the state file